multidataset_plugin.h

```diff
@@ -0,0 +1,65 @@
+#ifndef MULTIDATASET_PLUGIN_H
+#define MULTIDATASET_PLUGIN_H
+
+//#include <mpi.h>
+#include <stdlib.h>
+#include <string.h>
+#include <hdf5.h>
+#include "H5Timing.h"
+#define ENABLE_MULTIDATASET 0
+#define MULTIDATASET_DEFINE 1
+
+static int max_batch_size_g = 2;
+static int hdf5_method_g = -1;
```

Comment:
Instead of having these be global variables, how about moving them to an struct? It could be returned from the call to `init_multidataset()` and then passed to all functions needing the variables. That would guarantee that `init_` is actually called before any of the functions using its side-effect. It also avoids the use of global data which makes reasoning about thread-safety much harder.